### ISSUE TITLE
Fit.Template - auto dispose controls removed

### ISSUE THREAD

**Issue by Jemt** [Jemt/Fit.UI#40]
Using Fit.Template in a React like style where everything is re-created and re-rendered multiple times results in increased memory consumption over time if previously added UI controls are not either disposed or reused.

Allowing Fit.Template to automatically dispose controls that have been removed _after_ pushing changes to DOM would eliminate the need for the developer to manually handle memory usage.

However, we need to make sure that developers that wants to be switch different controls in and out of the view over time can do so without Fit.Template disposing their controls, so we should be able to disable the behaviour. I suggest that auto disposing is ON by default and can be disabled when creating the template instance:

Fit.Template = function(refreshable = false, autoDispose = true)

So basically Fit.Template keeps an index of all Fit.UI controls added, and after pushing updates it makes a diff between the new collection of UI controls and the old, and disposes all the old controls that are no longer part of the view.

**Comment by Jemt** [comment]
This task depends on #39 - all controls must be disposable for this to work reliably without memory leaks.

**Comment by Jemt** [comment]
It would make sense adding support for populating template using instances of Fit.UI controls now that Fit.Template is capable of disposing them.
